File: python/caesar_cypher/caesar_cypher.py
class CaesarCypher(object):
    alphabet_size = 26
    ord_points = {
        'upper': {
            'begin': ord('A'),
            'end': ord('Z')
        },
        'lower': {
            'begin': ord('a'),
            'end': ord('z')
        }
    }

    def shift_message(self, message, shift=3, left=False):
        shift = self.__normalize_shift_size(shift)
        if left:
            shift = -shift

        return ''.join([self.__moving(char, shift) for char in message])

    def __moving(self, char, shift):
        case = self.__get_case(char)
        if not case:
            return char

        position = ord(char) + shift
        if position > self.ord_points[case]['end']:
            position = position - self.alphabet_size
        elif position < self.ord_points[case]['begin']:
            position = position + self.alphabet_size

        return chr(position)

    def __get_case(self, char):
        if ord(char) >= self.ord_points['upper']['begin'] and\
                ord(char) <= self.ord_points['upper']['end']:
            return 'upper'
        elif ord(char) >= self.ord_points['lower']['begin'] and\
                ord(char) <= self.ord_points['lower']['end']:
            return 'lower'
        else:
            return False

    def __normalize_shift_size(self, shift):
        while shift > self.alphabet_size:
            shift = shift - self.alphabet_size

        return shift

File: php/caesar_cypher/caesar_cypher.php
<?php

class CaesarCypher
{
    const ALPHABET_SIZE = 26;
    const ORDER_POINTS = [
        'upper' => [
            'begin' => 65, // ord('A')
            'end' => 90 // ord('Z')
        ],
        'lower' => [
            'begin' => 97, // ord('a')
            'end' => 122 // ord('z')
        ]
    ];

    public function shiftMessage($message, $shift=3, $left=false)
    {
        $shift = $this->normalizeShiftSize($shift);
        if ($left) {
            $shift = -$shift;
        }

        $shifted = [];
        foreach (str_split($message) as $char) {
            $shifted[] = $this->moving($char, $shift);
        }

        return implode($shifted);
    }

    private function moving($char, $shift)
    {
        $case = $this->getCase($char);
        if (!$case) {
            return $char;
        }

        $position = ord($char) + $shift;
        if ($position > self::ORDER_POINTS[$case]['end']) {
            $position = $position - self::ALPHABET_SIZE;
        } elseif ($position < self::ORDER_POINTS[$case]['begin']) {
            $position = $position + self::ALPHABET_SIZE;
        }

        return chr($position);
    }

    private function getCase($char)
    {
        if (ord($char) >= self::ORDER_POINTS['upper']['begin'] &&
                ord($char) <= self::ORDER_POINTS['upper']['end']) {
            return 'upper';
        } elseif (ord($char) >= self::ORDER_POINTS['lower']['begin'] &&
                ord($char) <= self::ORDER_POINTS['lower']['end']) {
            return 'lower';
        } else {
            return false;
        }
    }

    private function normalizeShiftSize($shift)
    {
        while ($shift > self::ALPHABET_SIZE) {
            $shift = $shift - self::ALPHABET_SIZE;
        }

        return $shift;
    }
}
